src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -459,9 +459,7 @@ function assignOptionIds(
     (
       option: OptionDescriptor | ActionListItemDescriptor,
       optionIndex: number,
-    ) => {
-      option.id = `${comboBoxId}-${optionIndex}`;
```

Comment:
This code was a little odd and might be obscuring what it's actually doing. Map returns a new array, however, we seem to be mutating the option and not using the result of the map. Ideally, we would have used a forEach, for, for of, etc.